doc/howto/write_an_alphabet/index.md

```diff
@@ -0,0 +1,150 @@
+  that can be printed as part of a sequence. An Alphabet must also model seqan3::Semialphabet.
+- seqan3::WritableAlphabet additionally allows to assign the visual representation to an object of your Alphabet.
+
+# Step by step: Create your own alphabet
+
+In the alphabet tutorial we have calculated the GC content of a nucleotide sequence.
+Guanine and Cytosine are complementary nucleobases,
+which pair in a DNA molecule by building 3 hydrogen bonds. Adenine and Thymine pair with only 2 hydrogen bonds.
+As a consequence, we denote Guanine and Cytosine as strong (S) and Adenine and Thymine as weak (W) nucleobases.
+In this section we want to implement a seqan3::Alphabet that consists of the characters `S` and `W` to represent
+strong and weak nucleobases.
+
+Let's start with a simple struct that only holds the alphabet's rank value: 
+~~~cpp
+#include <seqan3/alphabet/concept.hpp>          // for seqan3::Alphabet concept checks
+
+struct dna2
+{
+    uint8_t rank{};
+};
+~~~
+
+If you want SeqAn's algorithms to accept it as an alphabet, you need to make sure that your type
+satisfies the requirements of the seqan3::Alphabet concept. A quick check can reveal that this is not the case:
+~~~cpp
+static_assert(seqan3::Alphabet<dna2>);          // compiler error
+~~~
+
+## Prerequisites
+
+A look at the documentation of seqan3::Alphabet will reveal that it is actually a refinement of other concepts, 
+more precisely seqan3::Semialphabet which in turn refines std::CopyConstructible and std::StrictTotallyOrdered. 
+Let's check those:
+
+~~~cpp
+static_assert(std::CopyConstructible<dna2>);    // OK
+static_assert(std::StrictTotallyOrdered<dna2>); // compiler error
+static_assert(seqan3::Semialphabet<dna2>);      // compiler error
+static_assert(seqan3::Alphabet<dna2>);          // compiler error
+~~~
+
+You should see that your type models only the std::CopyConstructible concept.
+Let's have a look at the documentation for std::StrictTotallyOrdered. Can you guess what it describes?
+
+It describes the requirements for types that are comparable via `<`, `<=`, `>` and `>=`.
+This is useful so that you can sort a text over the alphabet, for example.
+Additionally, in order to model std::StrictTotallyOrdered we need to make our alphabet
+[equality comparable](\ref std::EqualityComparable) by implementing the (in-)equality operators (as free functions).
+\snippet alphabet_custom_steps.cpp equality
+
+\assignment{Excercise}
+Implement the inequality operator (!=) for `dna2`.
+\endassignment
+\solution
+\snippet alphabet_custom_steps.cpp inequality
+\endsolution
+
+We see that our type now models std::EqualityComparable, which is a prerequisite of std::StrictTotallyOrdered.
+\snippet alphabet_custom_steps.cpp equality_comparable
+
+\assignment{Excercise}
+Implement the four compare operators *as free functions* and verify that your type models std::StrictTotallyOrdered.
+\endassignment
+\solution
+\snippet alphabet_custom_steps.cpp comparison
```

Comment:
please include the full solution. It is not too long yet and it should be copy-and-pastable.

For example, you have not shown any example with those functions *inside* the dna2 struct. I know this should be clear to most cpp developers but you never know